services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -25,17 +25,20 @@ import (
 	"os"
 	"path/filepath"
 	"syscall"
+	"time"
 )
 
 type metrics struct {
-	sizeOnDisk *metric.Gauge
+	sizeOnDisk          *metric.Gauge
+	indexLastUpdateTime *metric.Gauge
 }
 
 const blocksFilename = "blocks"
 
 func newMetrics(m metric.Factory) *metrics {
```

Comment:
if you are already touching this how about rename the functiont to generateBlockStorageMetrics ?